src/store.js

```diff
@@ -39,6 +39,28 @@ store.resetStore = function() {
 };
 
 /**
+ * Returns a flat array of voucher codes.
+ *
+ * @param trader
+ *   Currently not used. Will be needed when we support the storage of multiple traders.
+ * @returns {Array}
+ */
+store.getTraderVoucherList = function(trader) {
+    return this.trader.vouchers.map((v) => v.code);
+};
+
+/**
+ * Indicates whether any vouchers have been added whilst the user was offline.
+ *
+ * Returns false if any of the vouchers were recorded whilst the user was offline.
+ *
+ * @returns {Boolean}
+ */
+store.getVouchersOnlineStatus = function() {
+    return !!(this.trader.vouchers.reduce((v, e) => e.online, true));
```

Comment:
In what case would `this.trader.vouchers.reduce((v, e) => e.online, true)` not return the same value as the !! of it?

Will the reduce always return true or false or can it be undefined or a string? Why do you need to feed it an initial value of true?